src/ValueObject/ViewDataCollector.php

```diff
@@ -23,6 +23,17 @@ public function name(): string
         return $this->viewName;
     }
 
+    public function getName(): string
+    {
+        return $this->name();
+    }
+
+    public function getPath(): string
```

Comment:
@AJenbo, I went ahead and added the `getPath` method---however, I think what we are actually passing to this class **is** the path and not the  view *name*. This could potentially cause listeners that need to do something with the view name to fail

Reply:
I think it initially was when I implemented but I ended up reverting some of the changes, thanks for pointing that out.

It does look like it gets the view name in 2/3 cases, I'll fix the other one as well.

Reply:
I split this part into it's own PR: https://github.com/bladestan/bladestan/pull/159
It's already merged since I set it to auto-merge and forgot that Bladestan doesn't require views since there was no one else to do so :sweat_smile: @calebdw if you could check the code and test if it still solves the issue (with Telescope) that would be much appreciated.

The  issue caught by the test will require porting some Regex to PhpParser, this is absolutly something we want to do in general and this isn't the hardest one to do so it's a good one to work one next, I'll try and do so later today.

Reply:
@AJenbo, no worries! I was in church so couldn't check right away---but everything looks good and it works!